src/plans/solver_state.jl

```diff
@@ -211,58 +211,104 @@ Initialize the Functor to a set of keys, where the dictionary is initialized to
 be empty. Further, `once` determines whether more that one update per iteration
 are effective, otherwise only the first update is stored, all others are ignored.
 """
-mutable struct StoreStateAction <: AbstractStateAction
-    values::Dict{Symbol,<:Any}
-    keys::NTuple{N,Symbol} where {N}
+mutable struct StoreStateAction{TPS,TXS,TPI,TTI} <: AbstractStateAction
+    values::Dict{Symbol,Any}
+    keys::Vector{Symbol} # for values
+    point_values::TPS
+    tangent_values::TXS
+    point_init::TPI
```

Comment:
what is the init used for? Below it is accessed at some key? This should be documented maybe.

Reply:
This is here as a substitute to `has_storage` from the old solution, i.e. tells if a particular storage key has actually been saved already. I will add some docs later.

Reply:
looking forward to the docs then, since for now I do not understand what this field Is for.

Reply:
I've added some docs, let me know if they are sufficient.

Reply:
HM, I still do not understand, what I should pass to `point_values` and what to `point_inits` - both are named tuples but of what? I mean it seems this PR leaves another (huge?) PR to update all the rest – and for now I do not understand how to use this, so that makes a huge PR even more work.

So – what are the values what are the inits? Are they keys? Could they then be named keys? If none of them are keys, how to I find out which value is stored where? For example if two points are stored, which one is which field from state?

In short: Yes, all fields have a documentation, but they read like the comment `#this is a for loop` in front of a for loop for me, sorry :/

Reply:
also: tangent values and point values are documented twice, but different in the list.

Reply:
Sorry, I'm haven't seen this post before -- did the previous discussion make it a bit more clear?

Reply:
This is also not too old, no I still do not understand that well the constructors, but I hope I understand the NamedTuples by now. They key is a symbol, either the State field or a word with a capital letter like `:Gradient` (where for now the only other we have is `:Iterate`).